src/fpm_os.F90

```diff
@@ -102,4 +102,21 @@ subroutine c_f_character(rhs, lhs)
 
     end subroutine c_f_character
 
+    !> Determine the absolute from the relative path.
+    subroutine get_absolute_path(rel_path, abs_path, error)
+        character(len=*), intent(in) :: rel_path
+        character(len=:), allocatable, intent(out) :: abs_path
+        type(error_t), allocatable, intent(out) :: error
+        character(len=:), allocatable :: start_dir
+  
+        call get_current_directory(start_dir, error)
+        if (allocated(error)) return
+        call change_directory(rel_path, error)
+        if (allocated(error)) return
+        call get_current_directory(abs_path, error)
+        if (allocated(error)) return
+        call change_directory(start_dir, error)
+        if (allocated(error)) return
```

Comment:
That's an interesting approach. Do we have a function from the C stdlib we could call instead to avoid changing the directory. Resolving a relative path to an absolute one is definitely not easy, especially when symbolic links are involved.

Reply:
Yeah, `~` really doesn't work this way. (`exists('~')` is `.false.` and `change_directory` to it also doesn't work). Gotta try `getcwd`, `realpath` or `GetFullPathName`.

Reply:
In [this thread](https://fortran-lang.discourse.group/t/getting-a-full-path-name/4137/10) there are hints and some bits of code. It seems like some OS-specific operations are needed. i.e., use `GetFullPathName` on MS Windows and `getcwd` otherwise